app/models/article.rb

```diff
@@ -81,7 +72,7 @@ def source_text
 
   def self.delete_orphan_articles
     # don't delete orphan articles with contents
-    Article.delete_all("source_text IS NULL AND id NOT IN (select article_id from page_article_links)")
+    Article.where('source_text IS NULL AND id NOT IN (SELECT article_id FROM page_article_links)').destroy_all
```

Comment:
I think this has the same issue caveat as #4217 :
Project owners can upload canonical subjects via a spreadsheet, and we don't want to delete those if they are not used.

If the subject record was created via a spreadsheet upload, then the Article record will have a provenance of the filename they uploaded. Every subject that was created via the transcription process will have a nil provenance. So when we "clean up orphaned subjects" at page save, we need to only delete articles with a nil provenance and no links.
(In production 6000/200,000 have a provenance.)